src/Framework/Assert.php

```diff
@@ -1597,6 +1617,10 @@ public static function assertAttributeSame($expected, string $actualAttributeNam
      *
      * @throws ExpectationFailedException
      * @throws \SebastianBergmann\RecursionContext\InvalidArgumentException
+     *
+     * @psalm-template ExpectedType
+     * @psalm-param ExpectedType $expected
+     * @psalm-assert !=ExpectedType $actual
```

Comment:
```suggestion
```
I think this might be wrong - NotSame doesn't mean that the types have to be different.

Reply:
It doesn’t actually seem to have an effect. This should be a `RedundantCondition`: https://psalm.dev/8be275b3ec

Reply:
Oh it only complains when both are known string literals.

Reply:
The above code is now behaves as expected.

@bdsl the `!=ExpectedType` there means that Psalm expects the exact values to be different. In the case that it can determine the types are always different, it will emit a `RedundantCondition` or `RedundantConditionGivenDocblockType`, depending on the input.

Reply:
> @bdsl the `!=ExpectedType` there means that Psalm expects the exact values to be different. 

I got rid of this hint: as much as it may be about exact equality, I will keep myself on types-only for now. Please send a PR (against my branch) if you have a way to specifically test this.